libs/community/langchain_community/document_loaders/parsers/images.py

```diff
@@ -0,0 +1,149 @@
+import base64
+import html
+import io
+import logging
+from abc import abstractmethod
+from typing import TYPE_CHECKING, Iterator, Literal
+
+import numpy as np
+from langchain_core.documents import Document
+from langchain_core.language_models import BaseChatModel
+from langchain_core.messages import HumanMessage
+
+if TYPE_CHECKING:
+    from PIL.Image import Image
+
+from langchain_community.document_loaders.base import BaseBlobParser
+from langchain_community.document_loaders.blob_loaders import Blob
+
+logger = logging.getLogger(__name__)
+
+
+class ImageBlobParser(BaseBlobParser):
+    def __init__(
+        self,
+        *,
+        format: Literal["text", "markdown", "html"] = "text",
```

Comment:
This format is fairly surprising to see as part of the API -- but I think I'm OK with it.